core/src/main/java/org/frankframework/receivers/ReceiverAware.java

```diff
@@ -15,14 +15,16 @@
 */
 package org.frankframework.receivers;
 
+import org.frankframework.core.IListener;
+
 /**
  * When a listener implements this interface it will get a reference to it's
  * parent receiver.
  *
  * @author Jaco de Groot
  *
  */
-public interface ReceiverAware<M> {
+public interface ReceiverAware<M> extends IListener<M> {
```

Comment:
Dit voelt niet goed..

Reply:
Ik snap denk ik wel waar je vandaan komt met jouw bedenking, ik twijfelde er eerst ook over.

Maar alle `ReceiverAware` implementaties zijn listeners. De interface `IPortConnectedListener` dupliceerde deze interface, wat nu niet meer nodig is.
En door deze interface `... extends IListener<M>` te maken, zijn verschillende casts OK waarvan de compiler voorheen zei dat ze niet veilig waren.

Daarnaast zie ik in de JavaDoc staan
|  When a listener implements this interface ...

Dus ik denk dat de interface sowieso bedoelt was voor listeners, en niet voor andere classes die een link naar een receiver willen hebben.

Daarom dat ik denk dat het OK is.

Reply:
Hmm afbakenen dat deze dan alleen op een listener gezet kan worden...

Reply:
Hoe bedoel je dat?

Reply:
Ooit hier iets moois van maken dat men dit niet op een pipe kan gaan zetten en dan verwachten dat de receiver er ineens is :"P